packages/carousel/src/index.js

```diff
@@ -0,0 +1,228 @@
+  }
+
+  handleSwipedLeft(e, absX) {
+    const index =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+    const nextItemIndex = (index + 1) % this.state.items.length
+
+    this.setState({
+      internalCurrentItemIndex: nextItemIndex,
+      currentItemTranslate: 0,
+      isSwiping: false
+    })
+  }
+
+  handleSwipedRight(e, absX) {
+    const index =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+    const previousItemIndex =
+      (index + this.state.items.length - 1) % this.state.items.length
+
+    this.setState({
+      internalCurrentItemIndex: previousItemIndex,
+      currentItemTranslate: 0,
+      isSwiping: false
+    })
+  }
+
+  updateSelectedItem(selectedItemIndex) {
+    this.setState({
+      internalCurrentItemIndex: selectedItemIndex
+    })
+  }
+
+  mapItems() {
+    const { items } = this.state
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    const itemsToShow = []
+    items.forEach((item, index) => {
+      if (index <= currentItemIndex + 2) {
+        itemsToShow.push(
+          <span key={`item-${index}`}>
+            <PositionItem>{item}</PositionItem>
+          </span>
+        )
+      }
+    })
+
+    return itemsToShow
+  }
+
+  mapDots() {
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return this.state.items.map((item, mapIndex) => (
+      <Dot
+        key={mapIndex}
+        mx={2}
+        mt={2}
+        bg={mapIndex === currentItemIndex ? 'blue' : 'borderGray'}
+        onClick={this.updateSelectedItem.bind(this, mapIndex)}
+      />
+    ))
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      items: nextProps.children
+    })
+  }
+
+  render() {
+    const { currentItemTranslate, isSwiping } = this.state
+
+    const { shouldShowArrows } = this.props
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return (
+      <Box {...this.props}>
+        <OverflowContainer w={1} mb={3}>
+          {shouldShowArrows && (
+            <div>
+              <PositionedArrowLeftController
+                handleClick={this.handleSwipedRight}
+                iconDirection="chevronLeft"
+              />
+              <PositionedArrowRightController
+                handleClick={this.handleSwipedLeft}
+                iconDirection="chevronRight"
+              />
+            </div>
+          )}
+          <Swipeable
+            onSwipingLeft={this.handleSwipingLeft}
+            onSwipingRight={this.handleSwipingRight}
+            onSwipedLeft={this.handleSwipedLeft}
+            onSwipedRight={this.handleSwipedRight}
+            trackMouse
+          >
+            <SwiperWrap
+              translateX={-92 * currentItemIndex + currentItemTranslate + 4}
+              isSwiping={isSwiping}
+            >
+              {this.mapItems()}
```

Comment:
Method names like this tend to suggest that this should be its own component, e.g. `ItemsList`